Classes/JRHostsFile.m

```diff
@@ -4,12 +4,28 @@ @implementation JRHostsFile
 
 - (id)init
 {
+    return [self initWithFileName: [self defaultFileName]];
+};
+
+- (id)initWithFileName:(NSString*)fileName
+{
     self = [super init];
+
     if (self)
     {
-        // superclass successfully initialized, further
-        // initialization happens here ...
+        self.fileName = fileName;
     }
     return self;
+};
+
+- (NSString*) defaultFileName
+{
+    return @"/etc/hosts";
+};
+
+- (NSString*) getFileName
```

Comment:
I believe this isn't necessary and properties are automatically synthesized, so you already have a `.fileName` property on an instance and you don't want to reimplement another method to get it.